conan/tools/ros/rosenv.py

```diff
@@ -0,0 +1,49 @@
+import os
+from conan.api.output import Color
+from conan.tools.env import VirtualBuildEnv, Environment
+from conan.tools.files import save
+
+
+class ROSEnv:
+    """
+    Generator to serve as integration for Robot Operating System 2 development workspaces.
+    It generates a conanrosenv.sh file that when sources sets variables so the Conan
+    dependencies are found by CMake and the run environment is also set.
+
+    IMPORTANT: This generator should be used together with CMakeDeps and CMakeToolchain generators.
+    """
+
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+        self._variables = {}
```

Comment:
``self._variables`` doesn't need to be an object attribute, it can be a local variable inside ``generate()`` method